python/ray/experimental/client/server/server.py

```diff
@@ -17,35 +17,34 @@
 import time
 import inspect
 import json
-from ray.experimental.client import stash_api_for_tests, _set_server_api
 from ray.experimental.client.server.server_pickler import convert_from_arg
 from ray.experimental.client.server.server_pickler import dumps_from_server
 from ray.experimental.client.server.server_pickler import loads_from_client
-from ray.experimental.client.server.core_ray_api import RayServerAPI
 from ray.experimental.client.server.dataservicer import DataServicer
 from ray.experimental.client.server.logservicer import LogstreamServicer
 from ray.experimental.client.server.server_stubs import current_remote
+from ray._private.client_mode_hook import disable_client_hook
 
 logger = logging.getLogger(__name__)
 
 
 class RayletServicer(ray_client_pb2_grpc.RayletDriverServicer):
-    def __init__(self, test_mode=False):
+    def __init__(self):
         self.object_refs: Dict[str, Dict[bytes, ray.ObjectRef]] = defaultdict(
             dict)
         self.function_refs = {}
         self.actor_refs: Dict[bytes, ray.ActorHandle] = {}
         self.actor_owners: Dict[str, Set[bytes]] = defaultdict(set)
         self.registered_actor_classes = {}
-        self._test_mode = test_mode
         self._current_function_stub = None
 
     def ClusterInfo(self, request,
                     context=None) -> ray_client_pb2.ClusterInfoResponse:
         resp = ray_client_pb2.ClusterInfoResponse()
         resp.type = request.type
         if request.type == ray_client_pb2.ClusterInfoType.CLUSTER_RESOURCES:
-            resources = ray.cluster_resources()
+            with disable_client_hook():
```

Comment:
I'm a bit confused why this is needed--- isn't RAY_CLIENT_ENABLED False on the server?

Reply:
If you're running a server in a separate process than the client, sure. This becomes a noop. 

But in tests and in cases where it's trying to emulate `ray.init()` by way of running both locally, it's a bit more complicated.

That is an alternative for a follow-up -- spawn the server as a new process and handle that, with the right env changes -- but that's not how we've been running this whole time.